include/cantera/thermo/PengRobinson.h

```diff
@@ -0,0 +1,359 @@
+     *
+     *  @param species_i   Name of one species
+     *  @param species_j   Name of the other species
+     *  @param a0          constant term in the "a" expression [Pa-m^6/kmol^2]
+     *  @param a1          temperature-proportional term in the "a" expression
+     *      [Pa-m^6/kmol^2/K]
+     */
+    void setBinaryCoeffs(const std::string& species_i,
+                         const std::string& species_j, double a0, double a1);
+
+protected:
+    // Special functions inherited from MixtureFugacityTP
+    virtual double sresid() const;
+    virtual double hresid() const;
+
+	virtual double liquidVolEst(double TKelvin, double& pres) const;
+	virtual double densityCalc(double TKelvin, double pressure, int phase, double rhoguess);
+
+	virtual double densSpinodalLiquid() const;
+	virtual double densSpinodalGas() const;
+	virtual double pressureCalc(double TKelvin, double molarVol) const;
+	virtual double dpdVCalc(double TKelvin, double molarVol, double& presCalc) const;
+
+	// Special functions not inherited from MixtureFugacityTP
+
+	double daAlpha_dT() const;
+	double d2aAlpha_dT2() const;
+
+public:    
+
+    //! Calculate dpdV and dpdT at the current conditions
+    /*!
+     *  These are stored internally.
+     */
+    void pressureDerivatives() const;
+
+    //! Update the a and b parameters
+    /*!
+     *  The a and the b parameters depend on the mole fraction and the
+     *  temperature. This function updates the internal numbers based on the
+     *  state of the object.
+     */
+    virtual void updateMixingExpressions();
+
+    //! Calculate the a and the b parameters given the temperature
+    /*!
+     * This function doesn't change the internal state of the object, so it is a
+     * const function.  It does use the stored mole fractions in the object.
+     *
+     * @param temp  Temperature (TKelvin)
+     * @param aCalc (output)  Returns the a value
+     * @param bCalc (output)  Returns the b value.
+     */
+    void calculateAB(double& aCalc, double& bCalc, double& aAlpha) const;   
+
+    void calcCriticalConditions(double a, double b,double& pc, double& tc, double& vc) const;
+
+    //! Prepare variables and call the function to solve the cubic equation of state
+    int solveCubic(double T, double pres, double a, double b, double aAlpha,
+                     double Vroot[3]) const;
+protected:
+    //! Form of the temperature parameterization
+    /*!
+     *  0 = There is no temperature parameterization of a or b
+     *  1 = The a_ij parameter is a linear function of the temperature
+     */
+    int m_formTempParam;
+
+    //! Value of b in the equation of state
+    /*!
+     *  m_b_current is a function of the temperature and the mole fractions.
+     */
+    double m_b_current;
+
+    //! Value of a and alpha in the equation of state
+    /*!
+     *  m_aAlpha_current is a function of the temperature and the mole fractions. m_a_current depends only on the mole fractions.

```

Comment:
Is `m_aAlpha_current` always a function of T, or only if `a` is also a function of T?

Reply:
`m_aAlpha_current` is (`a*alpha`). Since `alpha` is temperature dependent,` m_aAlpha_current` is always temperature dependent.

Reply:
Is it possible for `m_a_current` to be a function of the temperature, if `a` is parameterized based on the temperature?

Reply:
'`a`'s are species-specific constants, and `m_a_current `only depends on the mole fractions. Both `a` and `m_a_current `are not dependent on the temperature. The temperature dependence is taken into account only through a parameter '`alpha`'